### PR TITLE
improve `lexers/shell.rb`

### PR DESCRIPTION
**Pull request by rindeal**
This patches adds support for Gentoo and Exherbo package build recipes. It also sets proper mime-type to shell files.

### CODE REVIEW

**Comment by pyrmont** [comment]
@rindeal Thanks for your work on this! It doesn't look like maintainers have the ability to push code to the relevant branch of your fork so apologies for not being able to make these changes myself.

To get the checks to pass, you need to update the MIME type section of `spec/lexers/shell_spec.rb` to something like this:

```diff
     it 'guesses by mimetype' do
-      assert_guess :mimetype => 'application/x-shellscript'
+      assert_guess :mimetype => 'text/x-sh'
+      assert_guess :mimetype => 'text/x-shellscript'
     end
```

In addition, if you could update the list of filename globs in `lib/rouge/lexers/shell.rb` so that it's a bit neater, that'd be great, too. Something like this:

```diff
       tag 'shell'
       aliases 'bash', 'zsh', 'ksh', 'sh'
-      filenames '*.sh', '*.bash', '*.zsh', '*.ksh',
-                '.bashrc', '.zshrc', '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD'
+      filenames '*.sh', '*.bash', '*.zsh', '*.ksh', '.bashrc', '.zshrc',
+                '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD', '*.ebuild',
+                '*.eclass', '*.exheres-0', '*.exlib'
```

Let me know if anything is unclear!

File: lib/rouge/lexers/shell.rb [hunk 1]
```diff
@@ -9,10 +9,12 @@ class Shell < RegexLexer
...
       tag 'shell'
       aliases 'bash', 'zsh', 'ksh', 'sh'
-      filenames '*.sh', '*.bash', '*.zsh', '*.ksh',
-                '.bashrc', '.zshrc', '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD'
+      filenames '*.sh', '*.bash', '*.zsh', '*.ksh', '.bashrc', '.zshrc',
+                '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD', '*.ebuild',
+                '*.eclass', '*.exheres-0', '*.exlib'
 
-      mimetypes 'application/x-sh', 'application/x-shellscript'
+      mimetypes 'application/x-sh', 'application/x-shellscript', 'text/x-sh',
```
Comment (pyrmont) [inline lib/rouge/lexers/shell.rb]: When I went looking for the correct MIME type for shell scripts, I didn't find much but it looked like `text` was correct. Why did you prefer doing it this way?

Comment (rindeal) [reply lib/rouge/lexers/shell.rb]: as the [commit message](https://github.com/rouge-ruby/rouge/pull/716/commits/761606f1c41f1752cedc4b06094252ce4020b649) says:

```
`file` utility uses `text/*` mime-types, while Mozilla uses `application/*`
```

Comment (rindeal) [reply lib/rouge/lexers/shell.rb]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types:

Screenshot_20190910_014620

Comment (pyrmont) [reply lib/rouge/lexers/shell.rb]: Ah, sweet. Sorry, missed that it was in there. 

One last thing: can you add tests for each of these MIME types to `spec/lexers/shell_spec.rb`?

File: lib/rouge/lexers/shell.rb [hunk 2]
```diff
@@ -9,10 +9,12 @@ class Shell < RegexLexer
 
       tag 'shell'
       aliases 'bash', 'zsh', 'ksh', 'sh'
-      filenames '*.sh', '*.bash', '*.zsh', '*.ksh',
-                '.bashrc', '.zshrc', '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD'
+      filenames '*.sh', '*.bash', '*.zsh', '*.ksh', '.bashrc', '.zshrc',
+                '.kshrc', '.profile', 'APKBUILD', 'PKGBUILD', '*.ebuild',
+                '*.eclass', '*.exheres-0', '*.exlib'
```
Comment (pyrmont) [inline lib/rouge/lexers/shell.rb]: Ugh—sorry, I must be half asleep. I should have asked for unit tests for the new file globs. Could you add these as well? These go in the same file as the unit tests for MIME types.

Comment (rindeal) [reply lib/rouge/lexers/shell.rb]: done that few minutes ago

Comment (pyrmont) [reply lib/rouge/lexers/shell.rb]: Oops! You beat me to it! Thanks!

Comment (rindeal) [reply lib/rouge/lexers/shell.rb]: https://github.com/rouge-ruby/rouge/pull/716/commits/df97237670d4903ae299c6c57d4b406dc2e017cf

**Comment by pyrmont** [comment]
@rindeal Hey, it took two years but we finally got this one closed! :tada: Thanks for contributing to Rouge :)

**Comment by rindeal** [comment]
Yes, thank you very much.

**Pull request merged**